lib/ae_mdw/db/name.ex

```diff
@@ -55,12 +55,21 @@ defmodule AeMdw.Db.Name do
     plain_name
   end
 
-  @spec ptr_resolve(state(), Blocks.block_index(), binary()) ::
+  @spec ptr_resolve(state(), Blocks.block_index(), binary(), binary()) ::
           {:ok, binary()} | {:error, :name_revoked}
-  def ptr_resolve(state, block_index, name_hash) do
-    with {:ok, account_id} <-
-           :aens.resolve_hash("account_pubkey", name_hash, ns_tree!(state, block_index)) do
-      {:ok, Validate.id(account_id)}
+  def ptr_resolve(state, block_index, name_hash, pointer_key) do
+    with {:ok, id} <-
+           :aens.resolve_hash(pointer_key, name_hash, ns_tree!(state, block_index)) do
+      Validate.id(id)
```

Comment:
`Validate.id/1` might return `{:error, Error.Id}`, which isn't what ptr_resolve returns here. Since we know it's a valid ID we don't need to validate, and we can use `:aeser_id.specialize` instead, right?